lib/active_merchant/billing/gateways/payeezy.rb

```diff
@@ -180,50 +232,63 @@ def add_soft_descriptors(params, options)
       end
 
       def commit(params, options)
-        url = if options[:integration]
-          integration_url
-        elsif test?
-          test_url
-        else
-          live_url
-        end
+        url = base_url(options) + endpoint(params)
 
         if transaction_id = params.delete(:transaction_id)
           url = "#{url}/#{transaction_id}"
         end
 
         begin
-          body = params.to_json
-          response = parse(ssl_post(url, body, headers(body)))
+          response = api_request(url, params)
         rescue ResponseError => e
           response = response_error(e.response.body)
         rescue JSON::ParserError
-          response = json_error(raw_response)
+          response = json_error(e.response.body)
         end
 
         Response.new(
           success_from(response),
           handle_message(response, success_from(response)),
           response,
           test: test?,
-          authorization: authorization_from(params, response),
+          authorization: success_from(response) ? authorization_from(params, response) : nil,
           avs_result: {code: response['avs']},
           cvv_result: response['cvv2'],
           error_code: error_code(response, success_from(response))
         )
       end
 
-      def success_from(response)
-        response['transaction_status'] == 'approved'
+      def base_url(options)
+        if options[:integration]
+          integration_url
+        elsif test?
+          test_url
+        else
+          live_url
+        end
       end
 
-      def authorization_from(params, response)
-        [
-          response['transaction_id'],
-          response['transaction_tag'],
-          params[:method],
-          (response['amount'] && response['amount'].to_i)
-        ].join('|')
+      def endpoint(params)
+        params[:ta_token].present? ? '/securitytokens' : '/transactions'
+      end
+
+      def api_request(url, params)
+        if params[:ta_token].present?
+          callback = ssl_request(:get, "#{url}?#{post_data(params)}", nil, {})
+          payload = callback[/{(?:\n|.)*}/]
+          parse(payload)
+        else
+          body = params.to_json
+          parse(ssl_post(url, body, headers(body)))
+        end
+      end
+
+      def post_data(params)
+        return nil unless params
+        params.to_query
```

Comment:
Whoa `.to_query` :mind_blown:
That said, I think we're trying to reduce dependency on ActiveSupport, so you might check out how some other adapters construct queries.

Reply:
+1 to using something other than `.to_query` purely from an Active Support dependency position.

Reply:
👍